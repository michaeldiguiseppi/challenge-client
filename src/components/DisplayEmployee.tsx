import React from "react";
import type { Employee } from "./types";

type Props = {
  employee: Employee;
};

const DisplayEmployee: React.FC<Props> = ({ employee }) => {
  const { name, picture, email } = employee;
  return (
    <div className="employee-display-detail">
      <div className="employee-display-header-photo">
        <img src={picture?.large} alt={`${name.first}-${name.last}`} />
      </div>
      <div className="employee-display-header-details">
        <div className="employee-display-header-name">
          {`${name.title} ${name.first} ${name.last}`}
        </div>
      </div>
      <div className="employee-display-contact-info">Contact: {email}</div>
    </div>
  );
};

export default DisplayEmployee;
