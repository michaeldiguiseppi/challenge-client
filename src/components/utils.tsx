import type { Employee } from "./types";

export const mockEmployees: Employee[] = [
  {
    name: {
      first: "Niclas",
      last: "Stornes",
      title: "Mr",
    },
    email: "niclas.stornes@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/men/78.jpg",
    },
    id: null,
  },
  {
    name: {
      first: "Erik",
      last: "Banks",
      title: "Mr",
    },
    email: "erik.banks@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/men/67.jpg",
      large: "https://randomuser.me/api/portraits/men/67.jpg",
      medium: "https://randomuser.me/api/portraits/med/men/67.jpg",
    },
    id: null,
  },
  {
    name: {
      first: "Andrea",
      last: "Crespo",
      title: "Mrs",
    },
    email: "andrea.crespo@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/women/41.jpg",
      large: "https://randomuser.me/api/portraits/women/41.jpg",
      medium: "https://randomuser.me/api/portraits/med/women/41.jpg",
    },
    id: null,
  },
];
