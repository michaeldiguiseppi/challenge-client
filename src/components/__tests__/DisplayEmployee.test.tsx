import React from "react";
import { render, cleanup } from "@testing-library/react";
import DisplayEmployee from "../DisplayEmployee";
import type { Employee } from "../types";
import { mockEmployees } from "../utils";

beforeEach(cleanup);

const employee: Employee = mockEmployees[1];

describe("<DisplayEmployee />", () => {
  it("renders without crashing", async () => {
    const { container } = render(<DisplayEmployee employee={employee} />);
    expect(container).toMatchSnapshot();
  });

  it("should render a large profile image for the selected employee", () => {
    const { getByAltText } = render(<DisplayEmployee employee={employee} />);
    getByAltText(/erik-banks/i);
  });

  it("should render the full name of the selected employee", () => {
    const { getByText } = render(<DisplayEmployee employee={employee} />);
    getByText(/erik banks/i);
  });

  it("should render the contact email for the selected employee", () => {
    const { getByText } = render(<DisplayEmployee employee={employee} />);
    getByText(/contact:/i);
    getByText(/erik.banks@example.com/i);
  });
});
