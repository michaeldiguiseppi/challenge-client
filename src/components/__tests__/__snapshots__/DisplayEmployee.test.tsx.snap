// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayEmployee /> renders without crashing 1`] = `
<div>
  <div
    class="employee-display-detail"
  >
    <div
      class="employee-display-header-photo"
    >
      <img
        alt="Erik-Banks"
        src="https://randomuser.me/api/portraits/men/67.jpg"
      />
    </div>
    <div
      class="employee-display-header-details"
    >
      <div
        class="employee-display-header-name"
      >
        Mr Erik Banks
      </div>
    </div>
    <div
      class="employee-display-contact-info"
    >
      Contact: 
      erik.banks@example.com
    </div>
  </div>
</div>
`;
